### PR TITLE
Moq test conversion

### PR DESCRIPTION
**Pull request by ngastelum-ms**

### CODE REVIEW

File: iothub/device/tests/IotHubDeviceClientTests.cs [hunk 1]
```diff
@@ -267,19 +267,17 @@ await deviceClient
                 .ConfigureAwait(false);
 
             // assert
-            await innerHandler
-                .Received()
-                .DisableMethodsAsync(Arg.Any<CancellationToken>())
-                .ConfigureAwait(false);
+            innerHandler
```
Comment (drwill-ms) [inline iothub/device/tests/IotHubDeviceClientTests.cs:270]: I'm not sure what Verify does.

Comment (drwill-ms) [reply iothub/device/tests/IotHubDeviceClientTests.cs:270]: Shouldn't there be a `Times` parameter or something?

File: iothub/device/tests/IotHubDeviceClientTests.cs [hunk 2]
```diff
@@ -1141,15 +1139,15 @@ public async Task IotHubDeviceClient_SendTelemetryAsync_Cancelled_ThrowsOperatio
...
             // We will setup the main handler which can be either MQTT or AMQP or HTTP handler to throw
             // a cancellation token expiry exception (OperationCancelledException) to ensure that we mimic when a token expires.
             mainProtocolHandler
-                .When(x => x.SendTelemetryAsync(Arg.Any<TelemetryMessage>(), Arg.Any<CancellationToken>()))
-                .Do(x => { throw new OperationCanceledException(); });
-
-            ErrorDelegatingHandler errorHandler = new ErrorDelegatingHandler(null, mainProtocolHandler);
+                .When(x => x.SendTelemetryAsync(It.IsAny<TelemetryMessage>(), It.IsAny<CancellationToken>()))
+                .Invoking(x => { throw new OperationCanceledException(); });
+
```
Comment (drwill-ms) [inline iothub/device/tests/IotHubDeviceClientTests.cs:1149]: extra whitespace here

File: iothub/device/tests/Pipeline/ErrorDelegatingHandlerTests.cs [hunk 3]
```diff
@@ -131,97 +130,78 @@ public async Task ErrorHandler_NonTransientErrorOccured_ChannelIsRecreated()
...
             //initial OpenAsync to emulate Gatekeeper behavior
             var cancellationToken = new CancellationToken();
             await sut.OpenAsync(cancellationToken).ConfigureAwait(false);
 
             //set initial operation result that throws
 
+            var message = new TelemetryMessage(new byte[0]);
             bool[] setup = { false };
-            mockSetup(innerHandler).Returns(ci =>
+            innerHandler.Setup(x => x.SendTelemetryAsync(message, It.IsAny<CancellationToken>())).Returns(() =>
```
Comment (drwill-ms) [inline iothub/device/tests/Pipeline/ErrorDelegatingHandlerTests.cs:165]:
```suggestion
            innerHandler
                .Setup(x => x.SendTelemetryAsync(message, It.IsAny<CancellationToken>()))
                .Returns(() =>
```

File: iothub/device/tests/IotHubDeviceClientTests.cs (hunk 1 above)
Comment (ngastelum-ms) [reply iothub/device/tests/IotHubDeviceClientTests.cs:270]: I think Verify defaults to at least once if no Times parameter is passed in

Comment (drwill-ms) [reply iothub/device/tests/IotHubDeviceClientTests.cs:270]: I'd rather the code be explicit.

File: iothub/device/tests/IotHubDeviceClientTests.cs [hunk 4]
```diff
@@ -1141,15 +1182,15 @@ public async Task IotHubDeviceClient_SendTelemetryAsync_Cancelled_ThrowsOperatio
...
 
-            var mainProtocolHandler = Substitute.For<IDelegatingHandler>();
+            var mainProtocolHandler = new Mock<IDelegatingHandler>();
 
             // We will setup the main handler which can be either MQTT or AMQP or HTTP handler to throw
             // a cancellation token expiry exception (OperationCancelledException) to ensure that we mimic when a token expires.
             mainProtocolHandler
-                .When(x => x.SendTelemetryAsync(Arg.Any<TelemetryMessage>(), Arg.Any<CancellationToken>()))
-                .Do(x => { throw new OperationCanceledException(); });
+                .Setup(x => x.SendTelemetryAsync(It.IsAny<TelemetryMessage>(), It.IsAny<CancellationToken>()))
```
Comment (ngastelum-ms) [inline iothub/device/tests/IotHubDeviceClientTests.cs:1190]: Not super confident that this change is correct

Comment (drwill-ms) [reply iothub/device/tests/IotHubDeviceClientTests.cs:1190]: Why `Invoking` and not `Returns`? What's the difference?

Comment (drwill-ms) [reply iothub/device/tests/IotHubDeviceClientTests.cs:1190]: https://nsubstitute.github.io/help/callbacks/

Comment (drwill-ms) [reply iothub/device/tests/IotHubDeviceClientTests.cs:1190]: I wonder if this works the same if you use `Returns` instead.

`SendTelemetryAsync` returns something (a Task), but more importantly, we're just throwing an exception here. Seems that the callback/no-return-value-method doesn't really apply here.

**Pull request merged**